jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Retainable.java

```diff
@@ -80,6 +93,15 @@ default boolean release()
         return true;
     }
 
+    /**
+     * <p>Get the retained count. This value is volatile and should only be used for informational/debugging purposes.</p>
+     * @return the retained count
+     */
+    default int getRetained()
+    {
+        return -1;
```

Comment:
0 sounds like a more sensible default, doesn't it?

Reply:
Maybe.  or throw Unsupported operation exception, as it should only be used be real retainables

Reply:
This is working for now and -1 is a clearly wrong value, which is better than 0 which is not.